components/director/internal/domain/label/service.go

```diff
@@ -0,0 +1,129 @@
+package label
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/pkg/jsonschema"
+	"github.com/pkg/errors"
+)
+
+//go:generate mockery -name=LabelRepository -output=automock -outpkg=automock -case=underscore
+type LabelRepository interface {
+	Upsert(ctx context.Context, label *model.Label) error
+	GetByKey(ctx context.Context, tenant string, objectType model.LabelableObject, objectID, key string) (*model.Label, error)
+}
+
+//go:generate mockery -name=LabelDefinitionRepository -output=automock -outpkg=automock -case=underscore
+type LabelDefinitionRepository interface {
+	Create(ctx context.Context, def model.LabelDefinition) error
+	Exists(ctx context.Context, tenant string, key string) (bool, error)
+	GetByKey(ctx context.Context, tenant string, key string) (*model.LabelDefinition, error)
+}
+
+//go:generate mockery -name=UIDService -output=automock -outpkg=automock -case=underscore
+type UIDService interface {
+	Generate() string
+}
+
+type labelUpsertService struct {
+	labelRepo           LabelRepository
+	labelDefinitionRepo LabelDefinitionRepository
+	uidService          UIDService
+}
+
+func NewLabelUpsertService(labelRepo LabelRepository, labelDefinitionRepo LabelDefinitionRepository, uidService UIDService) *labelUpsertService {
+	return &labelUpsertService{labelRepo: labelRepo, labelDefinitionRepo: labelDefinitionRepo, uidService: uidService}
+}
+
+func (s *labelUpsertService) UpsertMultipleLabels(ctx context.Context, tenant string, objectType model.LabelableObject, objectID string, labels map[string]interface{}) error {
+	for key, val := range labels {
+		err := s.UpsertLabel(ctx, tenant, &model.LabelInput{
+			Key:        key,
+			Value:      val,
+			ObjectID:   objectID,
+			ObjectType: objectType,
+		})
+		if err != nil {
+			return errors.Wrap(err, "while upserting multiple labels")
+		}
+	}
+
+	return nil
+}
+
+func (s *labelUpsertService) UpsertLabel(ctx context.Context, tenant string, labelInput *model.LabelInput) error {
+	exists, err := s.labelDefinitionRepo.Exists(ctx, tenant, labelInput.Key)
```

Comment:
in this method following interactions with DB happen:
- select to check if definition exists
- insert if the definition does not exist
- select label defintion

it looks that in every case there is one not needed call